volatility3/framework/plugins/layerwriter.py

```diff
@@ -38,6 +38,11 @@ def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]
                 default=False,
                 optional=True,
             ),
+            requirements.StringRequirement(
```

Comment:
Please also bear in mind, plugins should be UI agnostic.  This is a very specific option to get a particular preferred filename chosen for the output.  It pretty much only applies to the CLI because a web interface would have you download the output and choose the filename at the time you did that.  I really don't want to clutter plugins with cosmetic options that are catering to a single UI because that reinforces that there's only one UI anyone bothers with.  Luckily, this does use the FileHandler interface, so other UIs can gather and decide how to deal with files, but just as a reminder, the filename is just a preferred filename.

Reply:
Mh .. just to clarify, I'm not the author of this plugin and it wasn't my idea to add this option. What I did is to fix an option that is missing as the code below is expecting that.

Following the commits, it seems that, for some reason, you removed the LayerWriter plugin option in [Plugins: Make layerwiter more configurable](https://github.com/volatilityfoundation/volatility3/commit/47ff37b07b79a44a30249a243b095a0edfc1845a), but the option is still expected in the code. Not sure if you forgot to remove that as well or what happened there but it looks like something went wrong there.

Reply:
Totally understood, sorry if my wording was personalized.  I was responding to the idea and how it relates to what we have at the moment, not necessarily you in particular, I apologize.  I just started changing my mind as the review went on and by this point I felt quite strongly against the concept I think.  I didn't spot that the plugin expected an output. 

Having said that, since layerwriter can now accept and write out multiple layers at a time, a single output file wouldn't be right, so I don't think giving the user the choice of output name makes sense at all any more.  We should name each layer after the layer's `name` and write it out that way.  The user will then know what the file is finally called from the plugin output (although there's changes needed for that included in this PR I think) so can locate and rename it appropriately.  We don't really have any other easily retrievable information to put in the name (such as filename, since the lowest layer might not be a FileLayer and there is no unified way of extracting a source from it).